mithril-common/src/chain_reader/interface.rs

```diff
@@ -0,0 +1,18 @@
+use async_trait::async_trait;
+
+use crate::{entities::ChainPoint, StdResult};
+
+use super::ChainBlockNextAction;
+
+/// The trait that reads events to either:
+/// - read next block on the chain
+/// - rollback to another point in case of rollback
+/// - do nothing
+#[async_trait]
+pub trait ChainBlockReader {
+    /// Intersect the point of the chain with the given point
+    async fn intersect_point(&mut self, point: &ChainPoint) -> StdResult<()>;
```

Comment:
@falcucci this trait modification is problematic and it should be reverted IMO: the `intersect_point` is an implementation related function that should exist only in the `PallasChainReader`.

Thus, we should keep to the previous trait definition:
```rust
pub trait ChainBlockReader {
    /// Get the next chain block
    async fn get_next_chain_block(
        &mut self,
        point: &ChainPoint,
    ) -> StdResult<Option<ChainBlockNextAction>>;
}
```

Reply:
in the previous implementation we were intersecting a tip for every next chain block, which should be: you first intersect and then poll the blocks.

> the intersect_point is an implementation related function that should exist only in the PallasChainReader.

why? intersecting the tip isn't a pallas defined rule, it's part of the [chainsync](https://github.com/IntersectMBO/ouroboros-consensus/blob/e9cf1eeb8afce0d44ab4a1284ad8c065a74673a3/ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/MiniProtocol/ChainSync/Server.hs#L103) mini protocols consensus

Reply:
furthermore, this change makes it possible to test the `has_no_agency` scenario, which brings us the ability to manipulate the server agency state accordingly with our necessities as you can see [here](https://github.com/txpipe/mithril/blob/adab158c40af399bc6281e1d35936f709c5c1620/mithril-common/src/chain_reader/pallas_chain_reader.rs?plain=1#L323-L325)

Reply:
OK, let's rename the function in the trait with `set_chain_point` :+1: